src/internet_identity/src/storage/anchor.rs

```diff
@@ -15,6 +15,18 @@ pub struct Anchor {
     devices: Vec<Device>,
 }
 
+impl Device {
+    /// Applies the values of `device_data` to self while leaving the other fields intact.
+    pub fn apply_device_data(&mut self, device_data: DeviceData) {
```

Comment:
btw just my 2cts but I can easily see us forgetting to update this function if a field is added!

Reply:
Agreed, I'll add a test that verifies it by checking that `DeviceData::from` a `Device` that had data applied to it is equal to the data that was applied.